tests/contrib/fastapi/test_fastapi.py

```diff
@@ -34,6 +35,13 @@ def tracer():
     fastapi_unpatch()
 
 
+@pytest.fixture
+def test_spans(tracer):
+    container = TracerSpanContainer(tracer)
+    yield container
+    container.reset()
```

Comment:
I'm late to the party, but `test_spans` is a very confusing name to me.
I'm also wondering why we need another layer like `TracerSpanContainer`.
The more layer we add in testing the more it's likely we'll miss a problem.

Reply:
It was mostly added to aid assertions... before we had a better approach like snapshot testing.

I think most places where it is used can just be replaced by snapshot testing instead.

Reply:
Yeah totally agreed that the name isn't great. It gives a non-implementation specific way to retrieve the spans/traces that have been generated from the library which I think is good. Not sure what a better name for this fixture would be though 🤔 

As for having an additional layer like `TracerSpanContainer`, I don't see how we can avoid it without a bunch of duplication.

This discussion is worth having but probably a bit out of the scope of this PR as this is already the way that we do testing across the library and we're just making it consistent here.